thonny/editors.py

```diff
@@ -92,8 +92,15 @@ def update_file_type(self):
         if self._filename is None:
             self._code_view.set_file_type(None)
         else:
-            ext = self._filename.split(".")[-1].lower()
-            if ext in PYTHON_EXTENSIONS:
+            # Include backup files: *.py.bak, *.py.orig, etc
+            get_dots = len(re.findall(r'\.', self._filename))
+            ext = self._filename.split(".")[-get_dots].lower()
+
+            # Extract the 1st line to get the env
+            with open(self._filename, 'r') as shebang:
+                is_python_file = bool('python' in shebang.__next__().strip())
```

Comment:
`bool` is not required, as `in` operator already returns a `bool`.

I suggest `has_python_shebang` as the variable name, because the `ext in PYTHON_EXTENSIONS or is_python_file` part sounds like "an animal is zebra if it has black and white stripes or it is zebra" ;)

Reply:
Please also consider this file named "my-favorite-programs":

```
python
cat
less
ssh
```

Would you want to color it as Python code?